app/models/puppet_error.rb

```diff
@@ -0,0 +1,7 @@
+class PuppetError < MailNotification
+  scope :all_hosts, -> { PuppetError.includes(:user_mail_notifications).where(:user_mail_notifications => {:interval => 'Subscribe to all hosts'} ) }
+
+  def subscribe
```

Comment:
nitpick - `subscription_options` fits the meaning a bit better in my opinion. Calling mail_notification.subscribe sounds like it's subscribing an user to the mail notification.